src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/IProjectSystemCache.cs

```diff
@@ -107,5 +114,11 @@ public interface IProjectSystemCache
         /// Clears all project cache data.
         /// </summary>
         void Clear();
+
+        /// <summary>
+        /// Reset the dirty flag to 0 (is Not Dirty) if the flag was already set ( like TSL!!).
+        /// </summary>
+        /// <returns><code>true</code> if the cache was dirty before and <code>false</code> otherwise</returns>
+        bool TestResetDirtyFlag();
```

Comment:
Not sure I get the TSL reference here. The name seems confusing to me. GetAndResetDirtyFlag?